src/main/java/de/tum/in/www1/artemis/service/iris/IrisSessionService.java

```diff
@@ -129,8 +141,11 @@ public void checkRateLimit(IrisSession session, User user) {
      */
     @SuppressWarnings("unchecked")
     private <S extends IrisSession> IrisSubFeatureWrapper<S> getIrisSessionSubService(S session) {
-        if (session instanceof IrisChatSession chatSession) {
-            return (IrisSubFeatureWrapper<S>) new IrisSubFeatureWrapper<>(irisChatSessionService, chatSession);
+        if (session instanceof IrisExerciseChatSession chatSession) {
```

Comment:
This can now be done with a switch statement 🚀 
```java
return switch (session) {
            case IrisExerciseChatSession chatSession ->
                (IrisSubFeatureWrapper<S>) new IrisSubFeatureWrapper<>(irisExerciseChatSessionService, chatSession);
            case IrisCourseChatSession courseChatSession ->
                (IrisSubFeatureWrapper<S>) new IrisSubFeatureWrapper<>(irisCourseChatSessionService, courseChatSession);
            case IrisHestiaSession hestiaSession ->
                (IrisSubFeatureWrapper<S>) new IrisSubFeatureWrapper<>(irisHestiaSessionService, hestiaSession);
            case IrisCompetencyGenerationSession irisCompetencyGenerationSession ->
                (IrisSubFeatureWrapper<S>) new IrisSubFeatureWrapper<>(irisCompetencyGenerationSessionService, irisCompetencyGenerationSession);
            case null, default ->
                throw new BadRequestException("Unknown Iris session type " + session.getClass().getSimpleName());
        };
```